### DISCUSSION TITLE
How to limit file size and file type in direct upload?

### DISCUSSION THREAD

**Discussion by mad-coder-365** [aws/aws-sdk-ruby#2989] [Q&A]
How to limit file size and file type in direct upload? I am using @rails/activestorage also.

I couldn't find anything for Bucket policy.

I am trying to limit it so that users can only upload images with max size 2mb.

**Comment by mullermp** [comment]
I think your best bet is to check the size of the file's body and raise an error client side. I'm not sure if there's a server side validation for size, but if there is, that seems like a waste of network and possibly cost.
